test/integration/clusterthrottle_test.go

```diff
@@ -163,6 +164,57 @@ var _ = Describe("Clusterthrottle Test", func() {
 		})
 	})
 
+	When("Group Pods", func() {
+		var (
+			podPassedGroup    []*corev1.Pod
+			thr               *v1alpha1.ClusterThrottle
+			podThrottledGroup []*corev1.Pod
+		)
+		BeforeEach(func() {
+			thr = MustCreateClusterThrottle(ctx,
+				MakeClusterThrottle(throttleName).Selector(DefaultNs, throttleKey, throttleName).
+					ThresholdPod(4).
+					ThresholdCpu("4").
+					Obj(),
+			)
+
+			for i := 0; i < 2; i++ {
+				podPassedGroup = append(podPassedGroup, MustCreatePod(ctx, MakePod(DefaultNs, fmt.Sprintf("passed-pod%d", i), "100m").Annotation(groupNameAnnotation, "passed").Label(throttleKey, throttleName).Obj()))
+			}
+			for i := 0; i < 3; i++ {
+				podThrottledGroup = append(podThrottledGroup, MustCreatePod(ctx, MakePod(DefaultNs, fmt.Sprintf("throttled-pod%d", i), "100m").Annotation(groupNameAnnotation, "throttled").Label(throttleKey, throttleName).Obj()))
+			}
```

Comment:
I guess we need to wait for all pods in `podPassedGroup` to be scheduled before creating pods in `podThrottledGroup`.
Otherwise, pods in `podThrottledGroup` can be scheduled before `podPassedGroup`.

Reply:
I realized that I need to prevent all of group pods from reaching PreFilter before all group pods are created. To do this, we first create the group pods with schedulingGates and then remove the schedulingGates after creation is complete.